openmdao/docs/theory_manual/total_derivs/total_derivs_theory.rst

```diff
@@ -0,0 +1,227 @@
+
+It may look a bit odd to take derivatives of output values with respect to residual values, but Hwang and Martins showed that if you adopt a specific mathematical representation for the equations, then :math:`\left[\frac{do}{dr}\right]` contains all the total derivatives you actually want.
+Again we refer you to the paper on the `MAUD architecture`_ for the details, but lets work through an illustrative example.
+
+.. math::
+
+    x = 1,
+
+    y_1 = y_2^2,
+
+    e^{-y_1 y_2} - x y_1 = 0,
+
+    f = y_1^2 - y_2 + 3.
+
+In an OpenMDAO model, each single equation would be represented by components, and the components would be connected as shown in the XDSM diagram below.
+:code:`Discipline 2` is a different color because it would be implemented using an :ref:`ImplicitComponent <comp-type-3-implicitcomp>`.
+All of the red boxes would be built using :ref:`ExplicitComponent <comp-type-2-explicitcomp>`.
+
+.. figure:: xdsm/simple_example_xdsm.png
+    :align: center
+    :width: 75%
+
+    XDSM for the simple example model used to explain total derivative computation
+
+Even though the model is made up of both explicit and implicit calculations, internally OpenMDAO transforms all variables into an implicit form as follows:
+
+.. math::
+
+    r_{x} &= R_x(x) = x - 1 = 0,
+
+    r_{y_1} &= R_{y_1}(y_1, y_2) = y_1-y_2^2 = 0,
+
+    r_{y_2} &= R_{y_2}(x, y_1, y_2) = e^{-y_1 y_2} - x y_1 = 0,
+
+    r_{f} &= R_{f}(f, y_1, y_2) = f - y_1^2 - y_2 + 3 = 0.
+
+
+Now we can use the fully implicit form with the UDE to compute :math:`\left[\frac{do}{dr}\right]`, which would end up being
+
+.. math::
+
+    \left[ \frac{do}{dr}\right] =
+    \left[ \begin{array}{cccc}\frac{d x}{d r_{x}} &  \frac{d x}{d r_{y_1}} & \frac{d x}{d r_{y_2}} & \frac{d x}{d r_{f}} \\
+                              \frac{d y_1}{d r_{x}} &  \frac{d y_1}{d r_{y_1}} & \frac{d y_1}{d r_{y_2}} & \frac{d y_1}{d r_{f}} \\
+                              \frac{d y_2}{d r_{x}} &  \frac{d y_2}{d r_{y_1}} & \frac{d y_2}{d r_{y_2}} & \frac{d y_2}{d r_{f}} \\
+                              \frac{d f}{d r_{x}} &  \frac{d f}{d r_{y_1}} & \frac{d f}{d r_{y_2}} & \frac{d f}{d r_{f}} \end{array} \right] =
+    \left[ \begin{array}{ccc} I & 0 & 0 & 0\\
+                              \frac{d y_1}{d x} & \frac{d y_1}{d r_{y_1}} & \frac{d y_1}{d r_{y_2}} & 0\\
+                              \frac{d y_2}{d x} & \frac{d y_2}{d r_{y_1}} & \frac{d y_2}{d r_{y_2}} & 0 \\
+                              \frac{d f}{d x} & \frac{d f}{d r_{y_1}} & \frac{d f}{d r_{y_2}} & I \end{array} \right].
+
+
+Note that the total derivative an optimizer would need, :math:`\frac{d f}{d x}`, is contained in the last row of :math:`\left[\frac{do}{dr}\right]`.
+Since, for this simple one variable problem, that is the only derivative we actually need then we don't need to compute all of :math:`\left[\frac{do}{dr}\right]`.
```

Comment:
simple one variable problem -> simple, one-variable problem;
actually need then we -> actually need [comma] then we